src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md

```diff
@@ -522,21 +521,78 @@ In such a case, two files could have the following names:
 The user is free to choose any other label than `singleband` and
 `multiband`, as long as they are consistent across subjects and sessions.
 
-### REQUIRED gradient orientation information
-
-The REQUIRED gradient orientation information corresponding to a DWI acquisition
-MUST be stored using `[*_]dwi.bval` and `[*_]dwi.bvec` pairs of files.
-The `[*_]dwi.bval` and `[*_]dwi.bvec` files MAY be saved on any level of the directory structure
+### REQUIRED gradient strength and orientation information
+
+The gradient strength and orientation information corresponding to a DWI acquisition
+is REQUIRED, and therefore, MUST be stored in the structure.
+The gradient strength and orientation information stored in the structure MUST correspond
+to "*effective*" values (as in, the actual orientations and strengths applied by the scanner),
+rather than the "*requested*" values (as in, the original gradient table design that is
+configured into the scanner's MR protocol settings).
+
+The RECOMMENDED way to store gradient information is a `[*_]dwi.tsv` file corresponding
+to one or more DWI files.
+The format of this TSV file, and associated metadata, are described below.
+The gradient information MAY be stored using the `[*_]dwi.bval` and `[*_]dwi.bvec` pair
+of files.
+However, the `[*_]dwi.[bval|bvec]` files are DEPRECATED and will be
+phased out in a future release.
```

Comment:
Since this is a breaking change shouldn't this be under the 2.x consideration? if not perhaps the wording at present could be changed to support both the current required bval/bvec and add the tsv file.

Reply:
I see this PR more as a bugfix (and hence that takes precedence over the backward compatibility principle). Happy to see proposals on better wording though.

That said, with the current text both would be supported. TSV would take precedence over bval/bvec, and implicitly (but not explicitly), the tsv should be equivalent to bval/bvec. The only "breaking" change is lifting the mandate to have bvec/bval files (if you have the tsv)

Reply:
@satra did my reply address your concern? How would you address this issue?

Reply:
sorry too many things going on. my suggestion would be to continue to require bval and bvec till bids 2.0. regarding it being a bug. is it really a bug, or just that it forces a certain consideration that does not work in all situations. i.e. software have to be aware of the details of the spec. if the latter, that to  me is not a bug but an alienation problem, which can be addressed by making the tsv required/recommended moving forward. it's possible i'm not completely understanding the bug part of the spec.

Reply:
@oesteban I am inclined on giving precedence to bvecs/bvals reason being that if we give precedence to TSV only a very narrow set of mainstream toolboxes will be able to handle BIDS DWI. @arokem @jchoude

Reply:
I think mandating bvec/bval is the only reasonable option here, plenty of tools rely on them (for better or worse) to inform the arrangement of the acquisition. Giving a secondary tsv is much friendlier imo, and gives room to grow in the future.

Reply:
I've worked a deep rewriting I hope addressing these issues: https://github.com/bids-standard/bids-specification/pull/352/commits/6c73743815b596dbe8d5c129829b83ac1463d4f6

> regarding it being a bug. is it really a bug, or just that it forces a certain consideration that does not work in all situations. (@satra)

It is not rigorously a bug, especially for scanners that write voxel coordinates because that's the information that goes into the bval/bvec. However, some scanners write world coordinates and at this moment, that requires a conversion into bval/bvec. That means, we are not re-formatting (e.g., what you would hope for most -if not all- of the DICOM -> NIfTI step), you really need to recalculate these numbers.

> which can be addressed by making the tsv required/recommended moving forward. (@satra)

Yes, this is what this PR tries to establish. Let me know if it is unclear, and please suggest edits.

> i.e. software have to be aware of the details of the spec.

Currently, the bval/bvec information could be written in world coordinates. Software would definitely digest it (as it is a de-facto standard format) but give you very wrong results if the dataset is e.g., oblique (which is common with EPI). So current specs are pretty dangerous.
This PR uses the TSV header to specify the coordinate system of the gradients, making this consideration unambiguous (this would be "the bug" we are fixing).

Another way to put it. ATM, BIDS cannot *directly* write metadata about gradients if the scanner provides scanner coordinates (as opposed to voxel coordinates). In such a scenario, you either "preprocess" the gradients (with the risk of doing it wrongly) and convert them to voxel, or even worse, you encode the metadata as voxel coordinates, leave no trace that this problem has happened.

> I am inclined on giving precedence to bvecs/bvals reason being that if we give precedence to TSV only a very narrow set of mainstream toolboxes will be able to handle BIDS DWI. (@francopestilli)

and 

> I think mandating bvec/bval is the only reasonable option here, plenty of tools rely on them (for better or worse) to inform the arrangement of the acquisition. Giving a secondary tsv is much friendlier imo, and gives room to grow in the future. (@JAgho)

IMHO the risk of having "valid" (from the spec POV) bvecs/bvals in the wrong coordinate system (without the possibility of detecting the problem) completely outweighs playing nice with existing software. Also, the conversion is so straightforward (and now including an example in the spec), that I honestly can't see any reason not to discourage the bvecs/bvals format.

Reply:
Addendum:

re: bvec/bval remain mandatory - please check the conversations during the early days of this longstanding PR. Although now there's more information to support those initial decisions, I think the new information has only confirmed the adequacy of them (IMHO).

Reply:
BTW - how does this PR breaks backward compatibility? The deprecation is exactly to tolerate the utilization of the old feature while issuing a strong recommendation of the replacement (https://en.wikipedia.org/wiki/Deprecation) - precisely to avoid breaking changes.

I'd agree with removing the `.bvec`/`.bval` only the 2.0 release (which would indeed be the breaking change).

Reply:
a breaking change in the specification is one where existing tooling stops working as a result of the change. typically deletions of required elements in any schema are breaking changes. additions are also breaking changes if the new keys introduced are required.

Reply:
> a breaking change in the specification is one where existing tooling stops working as a result of the change.

Yup, and since .bval/.bvec will still be allowed, a dataset that currently works will continue working on the same tool. The requirement is to have at least one of the two options (.tsv or .bval/.bvec) - so if the .tsv is not present, the specification remains exactly the same.
The new format is RECOMMENDED, and will only be mandatory when the deprecation finishes. Possibly BIDS 2.0.

The only effect of this PR is that now a BIDS dataset can be written with a much better, unambiguous format for the gradients. And that's not breaking compatibility because scanner coordinates could not be encoded (without conversion to voxels) before.

Anyhow, besides I personally think this is not breaking compatibility, there's also the fact that we indeed discussed how this deprecation would happen when Matt first submitted the PR. It would be unfortunate that we revisited those decisions (for the reasons above, and because that will set some precedent as to maybe preempt future contributions).

My impression is that although this proposal has important substance changes (I particularly like @jchoude's summary above), we seem to be focusing more on the formal aspect of the deprecation (which has largely been discussed already).

Reply:
i completely agree that this sub discussion has to do with principles of BIDS rather than the substance of this PR. however, if this PR is merged software that worked prior to this could stop working. for example, wouldn't this mean that after this change none of the scripts using the FSL tools will work on new datasets unless someone did an appropriate conversion?

most software doesn't check on bids version for a dataset given that bids has not released 2.0. i think that's the fundamental question in this thread. this PR just happens to be an instance of something that brings this up. i don't want to hold up this PR, but it seems that there are other instances where such an issue could come into play. as bids expands this could happen at a modality specific level such as this, or the common principles level (there were a few inheritance issues that were pushed off to bids 2.0). 

thus this is a meta discussion for bids as a whole, not just this PR. but merging this PR, means we have accepted this route at least for one instance.

Reply:
> for example, wouldn't this mean that after this change none of the scripts using the FSL tools will work on new datasets unless someone did an appropriate conversion?

For existing datasets, not at all - this PR does not mandate the update from bvecs/bvals to tsv.
For future datasets, you can generate the bvecs/bvals from the tsv file. Making this step more explicit will make it harder for the user, but will also ensure data is preserved as it was generated by the scanner.

If someone is motivated and has the knowledge to implement a test within the BIDS validator to ensure that the tsv and bvec/bval are consistent, that would be really helpful for the DWI community.